pkg/crc/preflight/preflight_checks_linux.go

```diff
@@ -350,7 +350,16 @@ func checkDaemonSystemdSockets() error {
 		return err
 	}
 
-	return checkSystemdUnit(vsockUnitName, vsockUnit, true)
+	if err := checkSystemdUnit(vsockUnitName, vsockUnit, true); err != nil {
+		return err
+	}
+
+	if olderDaemonVersionRunning() {
+		if err := killDaemonProcess(); err != nil {
+			return err
+		}
+	}
+	return nil
```

Comment:
In linux I have to add it in this socket file check otherwise `cross-lint` error out not using `killDaemonProcess` function to linux. If we don't want to do that then might be we need to create a different file `preflight_check_nonlinux` and move this there.